src/libmongoc/tests/test-mongoc-client.c

```diff
@@ -4080,8 +4080,8 @@ test_mongoc_client_get_handshake_hello_response_pooled (void)
    handshake_sd = mongoc_client_get_handshake_description (
       client, monitor_sd->id, NULL /* opts */, &error);
    ASSERT_OR_PRINT (handshake_sd, error);
-   BSON_ASSERT (MONGOC_SERVER_UNKNOWN !=
-                mongoc_server_description_type (handshake_sd));
```

Comment:
> mongoc_server_description_type always returned a string, so was this original code a bug since MONGOC_SERVER_UNKNOWN is an enum value?

Yes, this was a bug since it was comparing the string to the enum value.

> Separate question, is there any reason libmongoc doesn't define constants for these strings? I know the enum type is private. Is this related to just not wanting to declare possible server types in any sort of public ABI?

I am not aware of any reason since the API predates me. But that seems plausible. SDAM server types are entrenched and long lived enough to where I would not be opposed to defining string constants if that would be useful for wrapping drivers. Feel free to file a ticket or PR to make that change.